anta/tests/bfd.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""
+BFD test functions
+"""
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Optional
+
+from anta.models import AntaCommand, AntaTest
+
+
+class VerifyBFDPeersHealth(AntaTest):
+    """
+    Verifies there is no IPv4 BFD peer in the down state, remote disc is not zero and last down should be above the threshold for all VRF.
+    Expected results:
+        * success: The test will pass if IPv4 BFD peers are not down, remote disc is not zero and last down above the defined threshold for all VRF.
+        * failure: The test will fail if IPv4 BFD peers are down, remote disc is zero and last down below the defined threshold for all VRF.
+    """
+
+    name = "VerifyBFDPeersHealth"
+    description = "Verifies there is no IPv4 BFD peer in the down state, remote disc is not zero and last down should be above the threshold for all VRF."
```

Comment:
I think we should mention that the last down threshold value is optional somehow.

Reply:
description = "Verifies there is no IPv4 BFD peer in the down state and discriminator value of the remote system is not zero for all VRF. BFD peer last down in hours is optional check which should be above the threshold for all VRF."

Reply:
Looks good, you can modify depending on how you decide to rename the `last_down` variable.